src/Client/OAuth2ClientInterface.php

```diff
@@ -45,6 +47,17 @@ public function redirect(array $scopes, array $options);
      */
     public function getAccessToken(array $options = []);
 
+    /**
+     * Refresh the given AccessToken.
+     *
+     * @param array $options Additional options that should be passed to the getAccessToken() of the underlying provider
+     *
+     * @return AccessToken|\League\OAuth2\Client\Token\AccessTokenInterface
+     *
+     * @throws IdentityProviderException If token cannot be fetched
+     */
+    public function refreshAccessToken(AccessTokenInterface $accessToken, array $options = []);
```

Comment:
For backwards-compatibility (so that we don't need to bump a new major version) - could you *not* actually add the method to the interface, but instead add it as documentation above the method? Here is an example: https://github.com/symfony/symfony/pull/36515/files#diff-a1c8b41afc0729b212c093eba2d2e82fdd13e22884fbc5fdf7575f3ad1fab0c6

I'm not sure if we can document the exact `$options` parameter with as much detail as we have here, so we may need to move that into the description of the method in that new format.

Reply:
Yeah no problem, I'm curious how adding to an interface breaks BC though? I thought adding things was fine, it's only when you move/change existing things that BC break?

Reply:
Very fair question :). If we supply an interface in this bundle, then we have to assume (even if it's not a normal use-case) that someone who uses this bundle may have created their own custom implementation (class) for this interface. If we add a new method in a minor release, suddenly their class explodes because it's missing 1 method from the interface. It's total nerdery, but you can go to http://symfony.com/bc to read about all the rules of what is and is not allowed for the purposes of backwards-compatibility in Symfony's codebase :).